### ISSUE TITLE
RuntimeError：indicies value must be public

### ISSUE THREAD

**Issue by Sparks1Fly** [secretflow/spu#116]
Hi,
I  ran my code on SPU-0.3.2b5 ,  the following error occurred:
`RuntimeError: what: \n\t[Enforce fail at libspu/kernel/hlo/utils.cc:42] (value.isPublic()). indicies value must be public.`
 Part of my code:
```       
        j=jnp.argmax(self.cache)
        temp=self.calc(j)
        self.cache=self.cache.at[j].set(temp)
```

There seems to be an error because of 'j'
Can I use 'j' as an array index in SPU?

**Comment by rivertalk** [comment]
Hi @Sparks1Fly, thanks for the question.

Currently SPU only support `Public indexing`, which means `j` must not be a secret value.
The `secret indexing` will be supported in the future but the perf is meant to be very slow.
 
Thanks,
Jin

**Comment by imwangyt** [comment]
Hi @6fj . I also got the same error occurred on SPU-0.3.2b5, when i ran jax.unique.
Is there any alternative? thanks~
my code:
```python
# unique
x = np.concatenate([np.arange(0,10), np.arange(5, 15)])
print(f'x = {x}')

fn = lambda x: jnp.unique(x, return_counts=True, size=15)
print(f'cpu out = {fn(x)}')

spu_fn = ppsim.sim_jax(sim, fn)
z = spu_fn(x)
print(f'spu out = {z}')
```

**Comment by 6fj** [comment]
Hi @imwangyt @Sparks1Fly 

At this moment, we are still working on this feature.

However, please work around temporaily by setting [reveal_secret_indices](https://www.secretflow.org.cn/docs/spu/en/reference/runtime_config.html#runtimeconfig) to be true in SPU runtime config. SPU runtime would reveal secret indices when this flag is on. Please don't open this flag in production settings however.

We would notify when this feature is ready.

**Comment by imwangyt** [comment]
> However, please work around temporaily by setting [reveal_secret_indices](https://www.secretflow.org.cn/docs/spu/en/reference/runtime_config.html#runtimeconfig) to be true in SPU runtime config. SPU runtime would reveal secret indices when this flag is on. Please don't open this flag in production settings however.

Hi, @6fj , thanks for your reply.
But, when not set the `return_index` etc., why does the `jax.unique`  use the indexing?
Can you explain a little or provide relevant information? Thank you very much~

**Comment by anakinxc** [comment]
Hi @imwangyt 

You can check jax.numpy.unique implementation here

https://jax.readthedocs.io/en/latest/_modules/jax/_src/numpy/setops.html

**Comment by imwangyt** [comment]
Hi @anakinxc , thanks for you reply.

Hi @6fj , when toggled `reveal_secret_indices`,  i got two questions:
1. is there some options to change the log level? 
2. the spu out seems to be incorrect?

my code: 
```python
config = spu_pb2.RuntimeConfig(protocol=spu_pb2.ProtocolKind.ABY3, field=spu_pb2.FieldType.FM64)
# config.enable_pphlo_profile = True
# config.enable_pphlo_trace = True
config.reveal_secret_indicies = True
sim = ppsim.Simulator(3, config)

x1d = np.arange(10, 0, -1)
y1d = np.arange(6, 16)
y1d[0] = 99
print(f'x1d = {x1d}')
print(f'y1d = {y1d}')

def fn(x, y):
    a = jnp.append(x, y)
    return jnp.unique(a, return_counts=True, size=a.size, fill_value=-1)

z = fn(x1d, y1d)
print(f'cpu out = {z}')

spu_fn = ppsim.sim_jax(sim, fn)
z = spu_fn(x1d,y1d)
print(f'spu out = {z}')
```
my jupyter notebook out:
```shell
x1d = [10  9  8  7  6  5  4  3  2  1]
y1d = [99  7  8  9 10 11 12 13 14 15]
cpu out = (Array([ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 99, -1,
       -1, -1, -1], dtype=int32), Array([1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0], dtype=int32))

[2023-03-17 09:49:43.286] [warning] [pphlo_executor.cc:433] only unstable sort is supported for secret returns.
[2023-03-17 09:49:43.296] [warning] [dynamic_slice.cc:69] Reveal 0th start index of DynamicSlice
...(too many lines)
[2023-03-17 09:49:43.335] [warning] [dynamic_slice.cc:37] Reveal 0th start index of DynamicUpdateSlice
[2023-03-17 09:49:43.335] [warning] [dynamic_slice.cc:37] Reveal 0th start index of DynamicUpdateSlice
spu out = (array([ 1312705073, -1312705048,          10, -1312705048,          10,
                15,  1312705073, -1312705048,  1312705073,  1312705073,
       -1312705048,  1312705073,          15,          10,  1312705073,
       -1312705048,          -1,          -1,          -1,          -1],
      dtype=int32), array([1, 1, 2, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 0, 0, 0, 0],
      dtype=int32))

```

**Comment by anakinxc** [comment]
Hi @imwangyt 

Thanks for the repo code. I can confirm there is an answer mismatch.

We are taking a look.

Thanks

**Comment by anakinxc** [comment]
Hi @imwangyt 

I have created a minimized repro code and spotted the bug.

The actual issue is different from the issues discussed in this thread, so let's track this bug in #117 

Thanks

**Comment by anakinxc** [comment]
Fixed in 0.3.2b9
